### PR TITLE
Add support for `print` command in local (command line `-d`) debugger

### LINKED ISSUE #73389: 'print' command in command line debugger does not show anything
### Godot version

4.0 RC2

### System information

Mac OS 13.1

### Issue description

When using Godot command line debugger (via executing 'Godot -d ...'), 'p' (alias of 'print') command does not show anything in terminal instead of a blank line, regardless the express is a string or a variable name, just like below.

```
debug> p 'a'

```

### Steps to reproduce

1. Run 'Godot -d project.godot' from command line
2. Add a line 'var a = 1/0' some where in a script
3. Save the script to trigger a break in command line debugger
4. Type "p 'foo'" in terminal
5. It should print out a blank line

### Minimal reproduction project

Any project will do.

### PR DESCRIPTION
**Pull request by wenqiangwang**
* *Bugsquad edit, fixes: https://github.com/godotengine/godot/issues/73389*

### CODE REVIEW

**Comment by wenqiangwang** [comment]
This is an attempt to reopen PR #73729 after the branch deleted

File: modules/gdscript/gdscript_editor.cpp
Comment (akien-mga) [inline modules/gdscript/gdscript_editor.cpp:433]: The indentation in this code is invalid, please make sure to only use tabs for indentation. You can run clang-format when committing code by setting up `pre-commit` with:
```
pip install pre-commit
pre-commit install
```

File: modules/gdscript/gdscript_editor.cpp
Comment (akien-mga) [inline modules/gdscript/gdscript_editor.cpp:443]: We ban the usage of the `auto` keyword in the Godot codebase (with a few exceptions in complex lambdas) as we prefer the types to be explicit.

File: core/debugger/local_debugger.cpp
Comment (dalexeev) [inline core/debugger/local_debugger.cpp:211]:
```suggestion
			if (line.find_char(' ') < 0) {
```

File: modules/gdscript/gdscript_editor.cpp
Comment (dalexeev) [inline modules/gdscript/gdscript_editor.cpp:450]: I'm not familiar with the Debugger, but it might make sense to use a more useful string representation:

```suggestion
            return return_val.get_construct_string();
```

File: modules/gdscript/gdscript_editor.cpp
Comment (dalexeev) [inline modules/gdscript/gdscript_editor.cpp:454]:
```suggestion
    return String();
```

**Comment by wenqiangwang** [comment]
Thank for the feedback, everyone. The comments are not resolved.

**Comment by wenqiangwang** [comment]
With the change, we can evaluate expressions in local debugger like in the screenshot below:

Screenshot 2024-09-19 223448

File: core/debugger/local_debugger.cpp
Comment (dalexeev) [inline core/debugger/local_debugger.cpp:212]:
```suggestion
				print_line("Usage: print <expr>");
```

or

```suggestion
				print_line("Usage: print <expression>");
```

File: modules/gdscript/gdscript_editor.cpp
Comment (wenqiangwang) [reply modules/gdscript/gdscript_editor.cpp]: Sorry about that. Fixed now

File: modules/gdscript/gdscript_editor.cpp
Comment (AThousandShips) [inline modules/gdscript/gdscript_editor.cpp:436]:
```suggestion
	for (const String &name : names) {
```
You swapped this one, they should match the type of the container

Comment (AThousandShips) [reply modules/gdscript/gdscript_editor.cpp]: No worries! It's easy to get confused when getting feedback

**Comment by wenqiangwang** [comment]
Can you please approve or give feedback?

**Comment by wenqiangwang** [comment]
@akien-mga, at the moment, do you think there is anything I can do to get this change in? Thanks.

**Comment by akien-mga** [comment]
You can squash the commits to make the PR ready to merge. But it still requires an approval for a maintainer, which needs some time. I plan to test it myself to confirm the fix, but I'm quite busy, so I can't say when I will do it.

**Review by dalexeev — APPROVED** [review]
Looks good to me. There are differences between GDScript and `Expression`, so it would be ideal to implement this separately (parsing and compiling individual GDScript expressions). But since we don't have a foundation for this, it might make sense to use `Expression` for now. Especially since the recently merged REPL (#97647) also uses `Expression`.

**Review by akien-mga — APPROVED** [review]
Looks good to me!

Sorry for the delay, I was on leave in October.

**Comment by wenqiangwang** [comment]
Thanks for looking, everyone!

**Comment by akien-mga** [comment]
And thanks for your patience with our slow review process :)

**Pull request merged**
